Index: tdebindings/korundum/bin/krubyinit.cpp
===================================================================
--- tdebindings.orig/korundum/bin/krubyinit.cpp
+++ tdebindings/korundum/bin/krubyinit.cpp
@@ -31,6 +31,42 @@
 // to detect what they were started with
 static const char* script_name = "krubyinit_app";
 
+/*#ifdef	__dilos__
+#include <tdeglobal.h>
+
+class TDEGlobalNetworkManager;
+class TDEGlobalNetworkManager;
+class TDEHardwareDevices;
+
+TDEGlobalNetworkManager *
+TDEInstance::networkManager()const
+{
+
+	return (NULL);
+}
+
+TDEHardwareDevices *
+TDEInstance::hardwareDevices()const
+{
+
+	return (NULL);
+}
+
+TDEGlobalNetworkManager *
+TDEGlobal::networkManager()
+{
+
+	return (NULL);
+}
+
+TDEHardwareDevices *
+TDEGlobal::hardwareDevices()
+{
+
+	return (NULL);
+}
+#endif*/	/* __dilos__ */
+
 int main(int argc, char **argv) {
      ruby_init();
      ruby_script((char*)script_name);
Index: tdebindings/smoke/tde/Makefile.am
===================================================================
--- tdebindings.orig/smoke/tde/Makefile.am
+++ tdebindings/smoke/tde/Makefile.am
@@ -1,3 +1,4 @@
+CPPFLAGS+= -D__TDE_HAVE_TDEHWLIB
 lib_LTLIBRARIES = libsmoketde.la
 
 # set the include path for X, qt and KDE
@@ -16,11 +17,15 @@ x_sources =  x_1.cpp x_2.cpp x_3.cpp x_4
    x_16.cpp x_17.cpp x_18.cpp x_19.cpp x_20.cpp
 
 # which sources should be compiled for perlqttde
-libsmoketde_la_SOURCES = $(x_sources) smokedata.cpp 
+libsmoketde_la_SOURCES = $(x_sources) smokedata.cpp dilos_smoke_tde.cpp
 
 $(x_sources) smokedata.cpp:
 	perl generate.pl
 
+dilos_smoke_tde.cpp:
+	cp -f $(top_srcdir)/debian/local/$@ $@
+
 CLEANFILES = $(libsmoketde_la_SOURCES)
 
 .NOTPARALLEL:
+
Index: tdebindings/admin/acinclude.m4.in
===================================================================
--- tdebindings.orig/admin/acinclude.m4.in
+++ tdebindings/admin/acinclude.m4.in
@@ -1586,7 +1586,7 @@ if AC_TRY_EVAL(kde_line); then
 	# if you're trying to debug this check and think it's incorrect,
 	# better check your installation. The check _is_ correct - your
 	# installation is not.
-	if test -f actest.cpp && grep klineedit actest.cpp > /dev/null; then
+	if test -f actest.cpp && grep -i klineedit actest.cpp > /dev/null; then
 		kde_cv_uic_plugins=yes
 	fi
 fi
Index: tdebindings/kjsembed/docs/embedding/simple-embed/embedview.ui
===================================================================
--- tdebindings.orig/kjsembed/docs/embedding/simple-embed/embedview.ui
+++ tdebindings/kjsembed/docs/embedding/simple-embed/embedview.ui
@@ -180,4 +180,7 @@
     <includehint>kpushbutton.h</includehint>
     <includehint>kpushbutton.h</includehint>
 </includehints>
+<includes>
+    <include>kpushbutton.h</include>
+</includes>
 </UI>
