Index: tdenetwork/ConfigureChecks.cmake
===================================================================
--- tdenetwork.orig/ConfigureChecks.cmake
+++ tdenetwork/ConfigureChecks.cmake
@@ -38,7 +38,10 @@ if( BUILD_KOPETE OR BUILD_KTALKD )
 endif( )
 
 if( WITH_SLP )
+  tde_save( CMAKE_REQUIRED_LIBRARIES )
+  set( CMAKE_REQUIRED_LIBRARIES socket )
   check_library_exists( slp "SLPOpen" "" HAVE_SLP )
+  tde_restore( CMAKE_REQUIRED_LIBRARIES )
   if( HAVE_SLP )
     set(LIB_SLP slp )
   else( )
Index: tdenetwork/libtdevnc/CMakeLists.txt
===================================================================
--- tdenetwork.orig/libtdevnc/CMakeLists.txt
+++ tdenetwork/libtdevnc/CMakeLists.txt
@@ -175,14 +175,20 @@ check_function_exists(vprintf         LI
 check_function_exists(mmap            LIBVNCSERVER_HAVE_MMAP)
 check_function_exists(fork            LIBVNCSERVER_HAVE_FORK)
 check_function_exists(ftime           LIBVNCSERVER_HAVE_FTIME)
+tde_save( CMAKE_REQUIRED_LIBRARIES )
+set( CMAKE_REQUIRED_LIBRARIES nsl )
 check_function_exists(gethostbyname   LIBVNCSERVER_HAVE_GETHOSTBYNAME)
+tde_restore( CMAKE_REQUIRED_LIBRARIES )
 check_function_exists(gethostname     LIBVNCSERVER_HAVE_GETHOSTNAME)
 check_function_exists(inet_ntoa       LIBVNCSERVER_HAVE_INET_NTOA)
 check_function_exists(memmove         LIBVNCSERVER_HAVE_MEMMOVE)
 check_function_exists(memset          LIBVNCSERVER_HAVE_MEMSET)
 check_function_exists(mkfifo          LIBVNCSERVER_HAVE_MKFIFO)
 check_function_exists(select          LIBVNCSERVER_HAVE_SELECT)
+tde_save( CMAKE_REQUIRED_LIBRARIES )
+set( CMAKE_REQUIRED_LIBRARIES socket )
 check_function_exists(socket          LIBVNCSERVER_HAVE_SOCKET)
+tde_restore( CMAKE_REQUIRED_LIBRARIES )
 check_function_exists(strchr          LIBVNCSERVER_HAVE_STRCHR)
 check_function_exists(strcspn         LIBVNCSERVER_HAVE_STRCSPN)
 check_function_exists(strdup          LIBVNCSERVER_HAVE_STRDUP)
Index: tdenetwork/cmake/modules/TDEMacros.cmake
===================================================================
--- tdenetwork.orig/cmake/modules/TDEMacros.cmake
+++ tdenetwork/cmake/modules/TDEMacros.cmake
@@ -2185,7 +2185,7 @@ macro( tde_setup_architecture_flags )
   check_cxx_compiler_flag( -fPIE HAVE_PIE_SUPPORT )
   if( HAVE_PIE_SUPPORT )
     set( TDE_PIE_CFLAGS -fPIE )
-    set( TDE_PIE_LDFLAGS -pie )
+#    set( TDE_PIE_LDFLAGS -pie )
   endif( HAVE_PIE_SUPPORT )
 endmacro( )
 
Index: tdenetwork/kopete/protocols/jabber/jingle/libjingle/talk/base/network.cc
===================================================================
--- tdenetwork.orig/kopete/protocols/jabber/jingle/libjingle/talk/base/network.cc
+++ tdenetwork/kopete/protocols/jabber/jingle/libjingle/talk/base/network.cc
@@ -42,6 +42,9 @@
 extern "C" {
 #include <sys/utsname.h>
 #include <sys/ioctl.h>
+#ifdef	__dilos__
+#include <sys/sockio.h>
+#endif	/* __dilos__ */
 #include <net/if.h>
 #include <unistd.h>
 #include <errno.h>
Index: tdenetwork/kopete/protocols/jabber/jingle/libjingle/talk/p2p/base/CMakeLists.txt
===================================================================
--- tdenetwork.orig/kopete/protocols/jabber/jingle/libjingle/talk/p2p/base/CMakeLists.txt
+++ tdenetwork/kopete/protocols/jabber/jingle/libjingle/talk/p2p/base/CMakeLists.txt
@@ -36,7 +36,7 @@ tde_add_executable( relayserver
   SOURCES
     relayserver.cc relayserver_main.cc
   LINK
-    cricketbase-static cricketp2pbase-static pthread
+    cricketbase-static cricketp2pbase-static pthread socket nsl
   DESTINATION ${BIN_INSTALL_DIR}
 )
 
@@ -47,6 +47,6 @@ tde_add_executable( stunserver
   SOURCES
     stunserver.cc stunserver_main.cc
   LINK
-    cricketbase-static cricketp2pbase-static pthread
+    cricketbase-static cricketp2pbase-static pthread socket nsl
   DESTINATION ${BIN_INSTALL_DIR}
 )
Index: tdenetwork/kppp/kpppconfig.h
===================================================================
--- tdenetwork.orig/kppp/kpppconfig.h
+++ tdenetwork/kppp/kpppconfig.h
@@ -26,7 +26,10 @@
 #ifndef _KPPPCONFIG_H_
 #define _KPPPCONFIG_H_
 
-#if defined(__svr4__)
+#if defined(__dilos__)
+#define STREAMS
+#define _XOPEN_SOURCE_EXTENDED 1
+#elif defined(__svr4__)
 #define STREAMS
 #define _XOPEN_SOURCE 1
 #define _XOPEN_SOURCE_EXTENDED 1
Index: tdenetwork/ktalkd/ktalkd/find_user.cpp
===================================================================
--- tdenetwork.orig/ktalkd/ktalkd/find_user.cpp
+++ tdenetwork/ktalkd/ktalkd/find_user.cpp
@@ -375,8 +375,8 @@ int find_user(char *name, char *tty, cha
 #ifdef USE_UT_HOST
                         if (Options.debug_mode) {
                             syslog(LOG_DEBUG, "I wanna this:%s", ttyFound);
-                            syslog(LOG_DEBUG, "ut_host=%s", ubuf.ut_host);
-                            syslog(LOG_DEBUG, "%s", ubuf.ut_line);
+                            syslog(LOG_DEBUG, "ut_host=%s", ubuf->ut_host);
+                            syslog(LOG_DEBUG, "%s", ubuf->ut_line);
                         }
 #endif
                         break;
Index: tdenetwork/ktalkd/mail.local/CMakeLists.txt
===================================================================
--- tdenetwork.orig/ktalkd/mail.local/CMakeLists.txt
+++ tdenetwork/ktalkd/mail.local/CMakeLists.txt
@@ -18,5 +18,6 @@ include_directories(
 
 tde_add_executable( mail.local
   SOURCES mail.local.c
+  LINK socket nsl
   DESTINATION ${BIN_INSTALL_DIR}
 )
Index: tdenetwork/lanbrowsing/lisa/CMakeLists.txt
===================================================================
--- tdenetwork.orig/lanbrowsing/lisa/CMakeLists.txt
+++ tdenetwork/lanbrowsing/lisa/CMakeLists.txt
@@ -26,7 +26,7 @@ install( FILES README DESTINATION ${DATA
 
 tde_add_executable( lisa
   SOURCES main.cpp
-  LINK lisa-static
+  LINK lisa-static socket nsl
   DESTINATION ${BIN_INSTALL_DIR}
 )
 
@@ -35,7 +35,7 @@ tde_add_executable( lisa
 
 tde_add_executable( reslisa
   SOURCES strictmain.cpp
-  LINK lisa-static
+  LINK lisa-static socket nsl
   DESTINATION ${BIN_INSTALL_DIR}
 )
 
Index: tdenetwork/krfb/ConfigureChecks.cmake
===================================================================
--- tdenetwork.orig/krfb/ConfigureChecks.cmake
+++ tdenetwork/krfb/ConfigureChecks.cmake
@@ -17,8 +17,11 @@ if( NOT XTST_FOUND )
   tde_message_fatal( "xtst is required, but was not found on your system" )
 endif( )
 
+tde_save( CMAKE_REQUIRED_LIBRARIES )
+set( CMAKE_REQUIRED_LIBRARIES socket nsl )
 check_function_exists( getnameinfo HAVE_GETNAMEINFO )
 check_function_exists( getifaddrs HAVE_GETIFADDRS )
+tde_restore( CMAKE_REQUIRED_LIBRARIES )
 check_include_file( sys/sockio.h HAVE_SYS_SOCKIO_H )
 
 check_struct_has_member(
