--- qmake/project.cpp.orig	2015-04-10 17:15:10.731091347 +0200
+++ qmake/project.cpp	2013-12-30 17:58:15.471703644 +0100
@@ -685,7 +685,7 @@
 //	    fprintf(stderr,"Current QT version number: " + ver + "\n");
 	    if (ver != "" && ver != test_version) {
 		ver = test_version;
-		fprintf(stderr,"Changed QT version number to " + test_version + "!\n");
+		fprintf(stderr,"Changed QT version number to %s!\n", test_version.ascii());
 	    }
 	}
     }
