Index: tdeadmin/kdat/ConfigureChecks.cmake
===================================================================
--- tdeadmin.orig/kdat/ConfigureChecks.cmake
+++ tdeadmin/kdat/ConfigureChecks.cmake
@@ -10,7 +10,7 @@ check_cxx_source_compiles("
     #include <sys/mtio.h>
     int main() {
       struct mtget tapeStatus;
-      return GMT_EOF ( tapeStatus.mt_gstat );
+      return GMT_EOF ( tapeStatus.mt_dsreg );
     }"
     HAVE_MTGET_GSTAT
 )
Index: tdeadmin/lilo-config/common/String.cc
===================================================================
--- tdeadmin.orig/lilo-config/common/String.cc
+++ tdeadmin/lilo-config/common/String.cc
@@ -27,7 +27,9 @@
 ** Bug reports and questions can be sent to kde-devel@kde.org
 */
 #define _GNU_SOURCE 1
+#ifndef	__dilos__
 #include <features.h>
+#endif	/* __dilos__ */
 #include <string.h>
 #include <string>
 #include "String.h"
Index: tdeadmin/lilo-config/common/Files.cc
===================================================================
--- tdeadmin.orig/lilo-config/common/Files.cc
+++ tdeadmin/lilo-config/common/Files.cc
@@ -35,8 +35,10 @@ StringList &Files::glob(String name, enu
 	glob_t gl;
 	int flags=0;
 	StringList *result;
+#ifndef	__dilos__
 	if(dotfiles)
 		flags |= GLOB_PERIOD;
+#endif	/* __dilos__ */
 	if(braces)
 		flags |= GLOB_BRACE;
 	::glob(name, flags, NULL, &gl);
Index: tdeadmin/lilo-config/common/lilo.cc
===================================================================
--- tdeadmin.orig/lilo-config/common/lilo.cc
+++ tdeadmin/lilo-config/common/lilo.cc
@@ -317,10 +317,15 @@ bool liloconf::probe()
 				char *tmp=tmpnam(NULL);
 				tmp=tmpnam(NULL);
 				mkdir(tmp, 0700);
+#ifdef	__dilos__
+				if(!mount(*it, tmp, MS_RDONLY|MS_NOSUID, "pcfs", NULL))
+					mp=tmp;
+#else	/* !__dilos__ */
 				if(!mount(*it, tmp, "msdos", MS_MGC_VAL|MS_RDONLY|MS_NOSUID|MS_NODEV|MS_NOEXEC, NULL))
 					mp=tmp;
 				else if(!mount(*it, mp, "vfat", MS_MGC_VAL|MS_RDONLY|MS_NOSUID|MS_NODEV|MS_NOEXEC, NULL))
 					mp=tmp;
+#endif	/* __dilos__ */
 			}
 			if(!mp.empty()) {
 				struct stat s;
