Index: arts/mcop/iomanager.cc
===================================================================
--- arts.orig/mcop/iomanager.cc
+++ arts/mcop/iomanager.cc
@@ -29,6 +29,9 @@
 #include "thread.h"
 #include <stdio.h>
 #include <fcntl.h>
+#ifdef	__dilos__
+#include <strings.h>
+#endif	/* __dilos__ */
 
 #ifdef HAVE_SYS_SELECT_H
 #include <sys/select.h>		// Needed on some systems.
Index: arts/mcop/CMakeLists.txt
===================================================================
--- arts.orig/mcop/CMakeLists.txt
+++ arts/mcop/CMakeLists.txt
@@ -67,6 +67,6 @@ set( ${target}_SRCS
 tde_add_library( ${target} SHARED
   SOURCES ${${target}_SRCS}
   VERSION 1.0.0
-  LINK ltdlc-static ${ESOUND_LIBRARIES}
+  LINK ltdlc-static ${ESOUND_LIBRARIES} socket nsl
   DESTINATION ${LIB_INSTALL_DIR}
 )
Index: arts/artsc/artsdsp.c
===================================================================
--- arts.orig/artsc/artsdsp.c
+++ arts/artsc/artsdsp.c
@@ -75,7 +75,7 @@ static int speed = 0;
 static int channels = 0;
 static int frags;
 
-#if defined(HAVE_IOCTL_INT_INT_DOTS)
+#if defined(HAVE_IOCTL_INT_INT_DOTS) || defined(__dilos__)
 typedef int ioctl_request_t;
 #elif defined(HAVE_IOCTL_INT_ULONG_DOTS)
 typedef unsigned long ioctl_request_t;
@@ -120,9 +120,12 @@ static orig_access_ptr orig_access;
 static int artsdsp_debug = 0;
 static int artsdsp_init = 0;
 
+#ifndef	__dilos__
 void *mmap(void  *start,  size_t length, int prot, int flags,
                      int fd, off_t offset);
 int munmap(void *start, size_t length);
+#endif	/* !__dilos__ */
+
 #define CHECK_INIT() if(!artsdsp_init) artsdsp_doinit();
 
 /*
